include/psa/update.h

```diff
@@ -0,0 +1,274 @@
+    uint32_t build;
+} psa_fwu_image_version_t;
+
+/**
+ * @brief The implementation-specific data in the component information
+ *        structure.
+ */
+typedef struct { /* implementation-defined type */ } psa_fwu_impl_info_t;
+
+/**
+ * @brief Information about the firmware store for a firmware component.
+ */
+typedef struct psa_fwu_component_info_t {
+    /// @brief State of the component.
+    uint8_t state;
+    /// @brief Error for second image when store state is REJECTED or FAILED.
+    psa_status_t error;
+    /// @brief Version of active image.
+    psa_fwu_image_version_t version;
+    /// @brief Maximum image size in bytes.
+    uint32_t max_size;
+    /// @brief Flags that describe extra information about the firmware
+    ///        component.
+    uint32_t flags;
+    /// @brief Implementation-defined image location.
+    uint32_t location;
+    /// @brief Reserved for implementation-specific usage.
+    psa_fwu_impl_info_t impl;
+} psa_fwu_component_info_t;
+
+
+/**
+ * @brief Retrieve the firmware store information for a specific firmware
+ *        component.
+ *
+ * @param component Firmware component for which information is requested.
+ * @param info      Output parameter for component information.
+ *
+ * @return Result status.
+ */
+psa_status_t psa_fwu_query(psa_fwu_component_t component,
+                           psa_fwu_component_info_t *info);
+
+/**
+ * @brief The READY state: the component is ready to start another update.
+ */
+#define PSA_FWU_READY 0u
+
+/**
+ * @brief The WRITING state: a new firmware image is being written to the
+ *        firmware store.
+ */
+#define PSA_FWU_WRITING 1u
+
+/**
+ * @brief The CANDIDATE state: a new firmware image is ready for installation.
+ */
+#define PSA_FWU_CANDIDATE 2u
+
+/**
+ * @brief The STAGED state: a new firmware image is queued for installation.
+ */
+#define PSA_FWU_STAGED 3u
+
+/**
+ * @brief The FAILED state: a firmware update has been cancelled or has failed.
+ */
+#define PSA_FWU_FAILED 4u
+
+/**
+ * @brief The TRIAL state: a new firmware image requires testing prior to
+ *        acceptance of the update.
+ */
+#define PSA_FWU_TRIAL 5u
+
+/**
+ * @brief The REJECTED state: a new firmware image has been rejected after
+ *        testing.
+ */
+#define PSA_FWU_REJECTED 6u
+
+/**
+ * @brief The UPDATED state: a firmware update has been successful, and the new
+ *        image is now active.
+ */
+#define PSA_FWU_UPDATED 7u
+
+/**
+ * @brief Flag to indicate whether the image data in the component staging area
+ *        is discarded at system reset.
+ */
+#define PSA_FWU_FLAG_VOLATILE_STAGING 0x00000001u
+
+/**
+ * @brief Flag to indicate whether a firmware component expects encrypted images
+ *        during an update.
+ */
+#define PSA_FWU_FLAG_ENCRYPTION 0x00000002u
+
+/**
+ * @brief The maximum permitted size for block in psa_fwu_write(), in bytes.
+ */
+#define PSA_FWU_MAX_BLOCK_SIZE /* implementation-specific value */
```

Comment:
This is related to the maximum transfer size, not the block size of the storage. For example, it might be the size of the available RAM buffer in the Service used to transfer the data to the Firmware Store.

Would `PSA_FWU_MAX_WRITE_SIZE` be a better name for this?

Reply:
Yeah, I think so. Also maybe worth extending the doxygen with pretty much what you commented above?

Reply:
The generated doxygen only includes the summary of the API description. I will rename this and provide better text for the full API definition.